scalafix-core/shared/src/main/scala/scalafix/v1/SemanticDoc.scala

```diff
@@ -0,0 +1,144 @@
+package scalafix.v1
+
+import java.nio.file.Files
+import java.util
+import scala.meta.io.Classpath
+import scala.meta.io.RelativePath
+import scala.collection.mutable.ListBuffer
+import scala.meta._
+import scala.meta.contrib.AssociatedComments
+import scalafix.internal.util.SymbolTable
+import scalafix.util.MatchingParens
+import scalafix.util.TokenList
+import scala.meta.internal.{semanticdb3 => s}
+import scalafix.internal.patch.DeprecatedSemanticdbIndex
+import scalafix.internal.v1._
+import scalafix.util.SemanticdbIndex
+
+trait SemanticContext
```

Comment:
Why is there a need in a trait without any methods?

Reply:
Semantic patches like `Patch.replaceSymbols` take an implicit evidence that you have access to semantic APIs. It has no methods since it's extended by both v1.SemanticDoc and v0.SemanticdbIndex and used in `Patch._` methods. Moved to separate file and added docstring.